src/main/kotlin/org/jetbrains/research/testspark/tools/TestProcessor.kt

```diff
@@ -234,7 +241,8 @@ class TestProcessor(
         frames.removeFirst()
 
         frames.forEach { frame ->
-            if (frame.contains(projectContext.classFQN!!)) {
+            // classFQN will be null for the top level function
+            if (projectContext.classFQN != null && frame.contains(projectContext.classFQN!!)) {
```

Comment:
`!!` should be removable in `projectContext.classFQN!!` since we explicitly checked for it.

Reply:
[SMARTCAST_IMPOSSIBLE] Smart cast to 'String' is impossible, because 'projectContext.classFQN' is a mutable property that could have been changed by this time